### PR TITLE
drivers: sensor: add api function for getting a sensor attribute

### LINKED ISSUE #26167: Extend the sensor API with function for getting the value of a sensor attribute
## Introduction

The Zephyr sensor API currently allows for setting attributes for sensor channels, but no way of retrieving the value of a sensor attribute.

### Problem description

Extend the sensor API with function for getting the value of a sensor attribute. Sensor attributes should not be write-only, as they provide means for exposing sensor-specific settings that might be of interest for the calling application (calibration values, limits, ...).

### Proposed change

Add a new, optional API function for retrieving the value of a sensor attribute.

### Proposed change (Detailed)

Please see PR #26166.

### Dependencies

None.

### Concerns and Unresolved Questions

None.

## Alternatives

Each sensor could implement a sensor-specific API for retrieving the values of sensor attributes.

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add an API function for getting the value of a sensor attribute and extend the existing sensor API test case to test it.

Fixes #26167.

### CODE REVIEW

File: tests/drivers/sensor/src/dummy_sensor.c [hunk 1]
```diff
@@ -111,6 +111,20 @@ int dummy_sensor_attr_set(struct device *dev,
 	return 0;
 }
 
+int dummy_sensor_attr_get(struct device *dev,
+		      enum sensor_channel chan,
+		      enum sensor_attribute attr,
```
Comment (gmarull) [inline tests/drivers/sensor/src/dummy_sensor.c:116]: It would probably be nice to use `attr` here.

**Review by gmarull — COMMENTED** [review]
LGTM, left a small comment on the test dummy driver.

File: tests/drivers/sensor/src/dummy_sensor.c (hunk 1 above)
Comment (henrikbrixandersen) [reply tests/drivers/sensor/src/dummy_sensor.c:116]: Agreed. I just reversed what `dummy_sensor_attr_set()` does (which does not use `attr`). I will implement a check for `attr` in both functions. Thank you for reviewing.

Comment (henrikbrixandersen) [reply tests/drivers/sensor/src/dummy_sensor.c:117]: @gmarull Please re-review.

File: tests/drivers/sensor/src/dummy_sensor.c [hunk 2]
```diff
@@ -104,13 +104,29 @@ int dummy_sensor_attr_set(struct device *dev,
...
+		      struct sensor_value *val)
+{
+	struct dummy_sensor_data *data = dev->driver_data;
+
+	if (chan == SENSOR_CHAN_PROX &&
+	    attr == SENSOR_ATTR_UPPER_THRESH) {
+		val->val1 = data->val[4].val1;
+		val->val2 = data->val[4].val2;
+	}
+	return 0;
```
Comment (gmarull) [inline tests/drivers/sensor/src/dummy_sensor.c:127]: If only `SENSOR_CHAN_PROX` and `SENSOR_ATTR_UPPER_THRESH` arguments are supported function should return 0 only for that case and `-ENOTSUP` for the rest (I guess).

Comment (henrikbrixandersen) [reply tests/drivers/sensor/src/dummy_sensor.c]: Fixed.

**Review by gmarull — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]
Seems like a relevant addition

**Review by MaureenHelm — APPROVED** [review]

**Comment by carlescufi** [comment]
@henrikbrixandersen are you comfortable with me merging this? The relevant stakeholders have seen it and this is a new addition instead of a stable API change, so if you remove the RFC I can merge it directly.

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen are you comfortable with me merging this? The relevant stakeholders have seen it and this is a new addition instead of a stable API change, so if you remove the RFC I can merge it directly.

@carlescufi Yes, certainly. Thank you for following up on it.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**